libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts

```diff
@@ -117,4 +150,46 @@ export class ApiTransactionDataAccessService {
   sendVerifyWebhook(app: App, payload: Prisma.AppTransactionUpdateInput) {
     return this.appWebhook.sendWebhook(app, { type: AppWebhookType.Verify, payload })
   }
+
+  async confirmSignature(
+    appTransactionId: string,
+    {
+      blockhash,
+      lastValidBlockHeight,
+      signature,
+    }: {
+      blockhash: string
+      lastValidBlockHeight: number
+      signature: string
+    },
+  ) {
+    this.logger.verbose(`confirmSignature: confirming ${signature}`)
+
+    const confirmed = await this.data.solana.confirmTransaction(
+      {
+        blockhash,
+        lastValidBlockHeight,
+        signature: signature as string,
+      },
+      Commitment.Finalized,
+    )
+    if (confirmed) {
+      this.logger.verbose(`confirmSignature: ${Commitment.Finalized} ${signature}`)
+      this.data.solana.connection
+        .getParsedTransaction(signature, 'finalized')
+        .then((solanaTransaction) =>
+          this.data.appTransaction.update({
+            where: { id: appTransactionId },
+            data: {
+              solanaFinalized: new Date(),
+              solanaTransaction: JSON.parse(JSON.stringify(solanaTransaction)),
+              status: AppTransactionStatus.Finalized,
+            },
+          }),
+        )
+        .then((res) => {
+          console.log('res', 'It is now finalized', { confirmed, res })
+        })
```

Comment:
Consider using async await for better readability

```js
const solanaTransaction = await this.data.solana.connection.getParsedTransaction(signature, 'finalized')
const res = await this.data.appTransaction.update({
  where: { id: appTransactionId },
  data: {
    solanaFinalized: new Date(),
    solanaTransaction: JSON.parse(JSON.stringify(solanaTransaction)),
    status: AppTransactionStatus.Finalized,
  },
})

console.log('res', 'It is now finalized', { confirmed, res })
  ```

Reply:
Thanks. Addressing this right now.

Also, this should not be a `console.log`: 
<img width="529" alt="image" src="https://user-images.githubusercontent.com/36491/171047508-0a875ae7-26de-43c5-b3d2-906834cd78ec.png">